LiteCore/RevTrees/VectorRecord.cc

```diff
@@ -404,7 +404,8 @@ namespace litecore {
     void VectorRecord::updateDocFlags() {
         // Take the local revision's flags, and add the Conflicted and Attachments flags
         // if any remote rev has them.
-        auto newDocFlags = _docFlags - DocumentFlags::kConflicted - DocumentFlags::kHasAttachments;
+        auto newDocFlags = _docFlags - DocumentFlags::kDeleted -
+            DocumentFlags::kConflicted - DocumentFlags::kHasAttachments;
```

Comment:
I really don't like this line.  I don't get how it is supposed to work in the first place.  At this point all you are doing is saving the Sync flag right?  So wouldn't this be more accurate as `_docFlags & DocumentFlags::kSynced` ?

Reply:
(Or at least, `_docFlags & ~(DocumentFlags::kDeleted | DocumentFlags::kConflicted | DocumentFlags::kHasAttachments` )

Reply:
First, I wasn't sure if will work as expected until I got to tested the behavior of using the minus operator on the enum. It produces the same result as your two suggestions. I will make change to `_docFlags & DocumentFlags::kSynced` which is more concise and understandable.

Reply:
`_docFlags & DocumentFlags::kSynced` actually produces boolean result.

Reply:
Whoa...I didn't know that enums would overload the minus operator that's really wild!!  I thought this was some kind of hack gone wrong.  That changes my thoughts on the matter but still either way would be ok.